clarify plaintext and associated data (#45)
juberti/cryptex: draft-ietf-avtcore-cryptex.md

## Encryption Procedure

The encryption procedure is identical to that of {{RFC3711}} except for the

> region to encrypt, which is as shown in the section above:
> 
>     Plaintext = CSRC identifiers (if used) || header extension || 
>          RTP payload || RTP padding (if used) || RTP pad count (if used).

Reviewer: ```suggestion
cipher inputs and the location of the encrypted data.  The plaintext input to
the cipher is as follows:

~~~
Plaintext = CSRC identifiers (if used) || header extension data || 
     RTP payload || RTP padding (if used) || RTP pad count (if used).
~~~

Here "header extension data" refers to the content of the RTP extension field,
excluding the first four bytes (the RFC 5285 extension header).  The first
`4*CC` bytes of the ciphertext are placed in the CSRC field of the RTP header.
The remainder of the ciphertext is the RTP payload of the encrypted packet.
```

## Encryption Procedure

The encryption procedure is identical to that of {{RFC3711}} except for the
cipher inputs and the location of the encrypted data.  The plaintext input to
the cipher is as follows:

~~~
Plaintext = CSRC identifiers (if used) || header extension data || 
     RTP payload || RTP padding (if used) || RTP pad count (if used).
~~~

Here "header extension data" refers to the content of the RTP extension field,

> excluding the first four bytes (the RFC 5285 extension header).  The first

Reviewer: ```suggestion
excluding the first four bytes (the RFC 8285 extension header).  The first
```

       |  0xC0 or 0xC2 |    0xDE       |           length=3            | |
     +>+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+ |
     | |            contributing source (CSRC) identifiers             | |
     | |                               ....                            | |
     | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
     | |                  RFC 8285 header extensions                   | |
     | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
     | |                          payload  ...                         | |
     | |                               +-------------------------------+ |
     | |                               | RTP padding   | RTP pad count | |
     +>+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<+
     | ~                     SRTP MKI (OPTIONAL)                       ~ |
     | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
     | :                 authentication tag (RECOMMENDED)              : |
     | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
     |                                                                   |

>      +- Encrypted Portions*                     Authenticated Portion ---+

Reviewer: * Added intro paragraph, `~~~` fencing, and tag/title annotation
* Changed from "encrypted/authenticated portion" to "Plaintext/AAD Input"
* Removed the MKI / auth tag, since they're not relevant here

extension header (the "defined by profile" field and the length) are considered
AAD, even though they are non-contiguous in the packet if CSRCs are present.

~~~
Associated Data: fixed header || extension header (if X=1)
~~~

Here "fixed header" refers to the 12-byte fixed portion of the RTP header, and

> "extension header" refers to the four-byte RFC 5285 extension header ("defined

Reviewer: ```suggestion
"extension header" refers to the four-byte RFC 8285 extension header ("defined
```

Obsolete reference.
